src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -1127,6 +1127,122 @@ TIP: Since it runs a graphical application, the task will never exit and run for
 you want to check manually some results via the graphical application and then exit the application.
 The task will then terminate normally.
 
+== Workflow storage and versioning
+
+The Workflow Catalog provides the storage and versioning of workflows inside Buckets. It also provides a searching functionality using a simple query language to fetch workflows based on particular parameters. The Workflow Catalog is accessible via a REST API, and a http://swagger.io/[Swagger] interface is provided to help interacting with the API.
+
+=== Bucket concept
+
+A Bucket is a collection of Workflows that can be shared between users. The default settings provides the following Buckets:
+
+* Catalog
+* Templates (which is a set of workflow samples)
+
+=== Workflow versioning
+
+The Workflow Catalog provides versioning of workflows. Changes on an existing workflow will trigger the creation of a new version. By default, the catalog will always return the latest version of the workflows. It is possible to search for specific workflows using a query string.
+
+==== Workflow Revision
+
+Getting a workflow revision will actually fetch the following Workflow's metadata:
+
+* The _id_ of the current Workflow
+* The _name_ (of the Workflow)
+* The _project_name_
+* All _variables_ of the workflow
+* The _created_at_ creation date of the Workflow
+* The _revision_id_ of the current Workflow
+* The _bucket_id_ of the Bucket which holds the current Workflow
+* All _generic_information_ used by the Workflow
+
+NOTE: When we fetch a workflow, we actually fetch the latest revision of that workflow.
+
+==== Searching for workflows
+
+A query language is provided to search through either:
+
+* Workflows (all latest Workflows versions)
+* Workflows Revisions (all the versions of a specific Workflow)
+
+One simple query expression can be the following clause:
+
+    name = "My First Workflow"
+
+Which will look for all Workflow that verify that clause.
+Multiple clauses can be combined using conjunctions such as `AND` (or `&&`) and `OR` (or `||`).
+Evaluation priority is respected through operator precedence and parenthesises so the following statement:
+
+    Search for all Workflows that uses exactly 4 CPUs
+
+
+Can be expressed as:
+
+    (variable.name="nbCPU" OR variable.name="CPU") AND variable.value="4"
```

Comment:
It looks strange to give an example that use different names for referencing a same concept. It gives the impression that the query language is there because we cannot well structure our workflows, etc.